File: src/main/java/com/example/demo/entity/PaymentScheduleSettings.java
package com.example.demo.entity;

import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

@Getter
@Setter
public class PaymentScheduleSettings {
    private Credit credit;
    private BigDecimal creditSum;
    private Integer numOfMonths;
    private boolean differential;//false - аннуитетный, true - дифференцированный
}

File: frontend/src/app/entities/paymentScheduleSettings.ts
import { Credit } from "./credit";
import { CreditOffer } from "./creditOffer";

export class PaymentScheduleSettings{
    creditOffer:CreditOffer;
    credit:Credit;
    creditSum:number;
    numOfMonths:number;
    differential:boolean;
    public constructor(offer:CreditOffer,credit:Credit,creditSum:number,numOfMonths:number,diff:boolean){
        this.creditOffer = offer;
        this.credit = credit;
        this.creditSum = creditSum;
        this.numOfMonths = numOfMonths;
        this.differential = diff;
    }
}